src/sentry/incidents/logic.py

```diff
@@ -992,63 +992,56 @@ def trigger_incident_triggers(incident):
                 )
 
 
-def sort_by_priority_list(incident_triggers):
+def sort_by_priority_list(triggers: list[AlertRuleTrigger]) -> list[AlertRuleTrigger]:
     priority_dict = {
-        (CRITICAL_TRIGGER_LABEL, TriggerStatus.ACTIVE.value): 0,
-        (WARNING_TRIGGER_LABEL, TriggerStatus.ACTIVE.value): 1,
-        (WARNING_TRIGGER_LABEL, TriggerStatus.RESOLVED.value): 2,
-        (CRITICAL_TRIGGER_LABEL, TriggerStatus.RESOLVED.value): 3,
+        WARNING_TRIGGER_LABEL: 0,
+        CRITICAL_TRIGGER_LABEL: 1,
     }
     return sorted(
-        incident_triggers,
-        key=lambda t: priority_dict.get(
-            (t.alert_rule_trigger.label, t.status), len(incident_triggers) + t.id
-        ),
+        triggers,
+        key=lambda t: priority_dict.get(t.label, len(triggers) + t.id),
     )
 
 
-def prioritize_actions(incident_triggers):
+def prioritize_actions(triggers: list[AlertRuleTrigger]) -> list[AlertRuleTriggerAction]:
     """
-    Function that given an input array of incident_triggers, prioritizes those incident_triggers
-    based on the label of related alert_rule_trigger and their TriggerStatus, and then re-orders
-    actions based on that ordering
+    Function that given an input array of AlertRuleTriggers, prioritizes those triggers
+    based on their label, and then re-orders actions based on that ordering
     Inputs:
-        * incident_triggers: Array of instances of `IncidentTrigger`
+        * triggers: Array of instances of `AlertRuleTrigger`
     Returns:
         List of instances of `AlertRuleTriggerAction` that are ordered according to the ordering
-        of related prioritized instances of `IncidentTriggers`
+        of related prioritized instances of `AlertRuleTrigger`
     """
     actions = list(
-        AlertRuleTriggerAction.objects.filter(
-            alert_rule_trigger__in=[it.alert_rule_trigger for it in incident_triggers]
-        ).select_related("alert_rule_trigger")
+        AlertRuleTriggerAction.objects.filter(alert_rule_trigger__in=triggers).select_related(
+            "alert_rule_trigger"
+        )
     )
 
-    incident_triggers = sort_by_priority_list(incident_triggers=incident_triggers)
-    incident_triggers_dict = {
-        it.alert_rule_trigger.id: idx for idx, it in enumerate(incident_triggers)
-    }
+    triggers = sort_by_priority_list(triggers=triggers)
+    triggers_dict = {t.id: idx for idx, t in enumerate(triggers)}
 
     sorted_actions = sorted(
         actions,
-        key=lambda action: incident_triggers_dict.get(
+        key=lambda action: triggers_dict.get(
             action.alert_rule_trigger.id, len(actions) + action.id
         ),
     )
     return sorted_actions
 
 
-def deduplicate_trigger_actions(triggers):
+def deduplicate_trigger_actions(triggers: list[AlertRuleTrigger]) -> list[AlertRuleTriggerAction]:
```

Comment:
We are preferring warning triggers over critical triggers now because with these changes, warning trigger actions will always send the same info as critical actions + more. We want to send more information if we have to choose.